file.go

```diff
@@ -1,37 +1,121 @@
 package jcsv
 
-import "os"
+import (
+	"encoding/csv"
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"os"
+)
 
 type file struct {
 	// TODO: define variables if needed
+	JSONFormat map[string]interface{}
+	CSVFormat  struct {
+		Data [][]string
+		//if "headers==true", data[0] will contain headers
+		HasHeaders bool
+	}
 }
 
+func (f *file) initialize() {
+	f.JSONFormat = nil
+	f.CSVFormat.Data = nil
+	f.CSVFormat.HasHeaders = false
+}
 func ParseJsonFile(path string) (file, error) {
 	// TODO: open and read the given file into your `file` object
-	return file{}, nil
+	jsonFile, err := os.Open(path)
+	var myFile file
+	myFile.initialize()
+	if err != nil {
```

Comment:
You are doing other stuff before handling a possible error. Why?

Reply:
The function returns file object so if error occurred, err will not nil then it will return the default file object created before if.

Reply:
You can return an unnamed struct like this: `return file{}, err`.

Reply:
all right, I will resolve this.